### PR TITLE
Consul Integration

### PR DESCRIPTION
**Pull request by TheConnMan**
This PR adds Consul host status counts, service status counts, and agent metrics.

### CODE REVIEW

File: src/collectors/consul/ConsulCollector.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        try:
+            metrics = json.load(urllib2.urlopen(url + '/v1/agent/metrics'))
+
+            for gauge in metrics['Gauges']:
+                self.publish(self.trimmed_metric_name(
+                    gauge['Name']), gauge['Value'])
+
+            for counter in metrics['Counters']:
+                self.publish(
+                    'counter.' + self.trimmed_metric_name(counter['Name']), counter['Count'])
```
Comment (TheConnMan) [inline src/collectors/consul/ConsulCollector.py:93]: `counters.` is added to make this easily capturable in an ACS config. All these metrics are counters.

**Review by ibilous — APPROVED** [review]

**Comment by rclagett** [comment]
@TheConnMan Should this be a merge request into `dev` instead of `master`?

File: src/collectors/consul/ConsulCollector.py (hunk 1 above)
Comment (Grumblesmith) [reply src/collectors/consul/ConsulCollector.py:93]: These metrics would come through as `consul.counter.foo`?

Comment (TheConnMan) [reply src/collectors/consul/ConsulCollector.py:93]: Yes.

**Review by Grumblesmith — APPROVED** [review]

**Review by rclagett — APPROVED** [review]

**Pull request merged**